java/src/jmri/jmrit/signalling/SignallingSourcePanel.java

```diff
@@ -180,7 +169,10 @@ public void propertyChange(java.beans.PropertyChangeEvent e) {
             boolean newValue = (Boolean) e.getNewValue();
             discoverPairs.setEnabled(newValue);
         }
-        //log.debug("SSP 179 Event: {}", e.getPropertyName());
+        log.debug("SSP 173 Event: {}; Source: {}", e.getPropertyName(), e.toString()); // doesn't get notified, newDestination
+        if (e.getPropertyName().equals("Frame.active")) { // a blunt way of redrawing the Pairs table
```

Comment:
Actually, digging into the code, you want to listen to changes in the "length" property of the SignalMastManager to trigger table model updates.